### PR TITLE
[O11YP-355] add in filtering features to pagerduty_nagios.pl

### PR DESCRIPTION
**Pull request by cswingler**
Add in some features to do the following:

* Filter out Nagios environment variables we don't care about
* Filter out any environment variables that are set to an empty string, or contain an error message
* Add an attribute that declares the hostname of the Nagios system that generated the alert

Duplicate of https://github.com/PagerDuty/pagerduty-nagios-pl/pull/13 since that project looks  very dead.

### CODE REVIEW

File: pagerduty_nagios.pl [hunk 1]
```diff
@@ -37,6 +36,67 @@
...
+# that we DO care about and everything else is just filtered out.
+# This is to prevent events from sending all 200+ fields that Nagios generates
+# over to your phone when a page goes out.
+# SOME OF these are defined in https://github.com/NagiosEnterprises/nagioscore/blob/master/include/macros.h
+# (replace MACRO_ with NAGIOS_)
+# Not all. Welcome to Nagios, enjoy your stay.
+# Note that this does a SUBSTRING MATCH, any environment variable that contains
+# these strings will be dropped.
+# Please keep this list sorted.
+# Note that any unpopulated fields will be dropped and Anot sent to Pagerduty.
```
Comment (r4v5) [inline pagerduty_nagios.pl]:
```suggestion
# Note that any unpopulated fields will be dropped and not sent to Pagerduty.
```

File: pagerduty_nagios.pl [hunk 2]
```diff
@@ -107,6 +167,85 @@ =head1 OPTIONS
...
+}
+
+#** @method public filter_fields_for_service_type (%event_hash)
+# Given a %event_hash, remove all the fields that are in @delete_when_service
+# Returns the filtered hash.
+#*
+sub filter_fields_for_service_type {
+	my (%events_hash) = @_;
+	if ($events_hash{"NAGIOS_pd_nagios_object"} ne "service") {
+		print STDERR "Expected this to be a service, it's $events_hash{'NAGIOS_pd_nagios_object'}. Returning untoutched.\n" if $opt_verbose;
```
Comment (r4v5) [inline pagerduty_nagios.pl]:
```suggestion
		print STDERR "Expected this to be a service, it's $events_hash{'NAGIOS_pd_nagios_object'}. Returning untouched.\n" if $opt_verbose;
```

File: pagerduty_nagios.pl [hunk 3]
```diff
@@ -107,6 +167,85 @@ =head1 OPTIONS
 my $opt_verbose;
 my $opt_proxy;
 
+#** @method public is_variable_allowed ($variable_name, $variable_value)
+# Given an variable name, see if we want to allow it to be put in a PagerDuty payload.
+# Returns 1 if it should be carried over, 0 otherwise
+#*
+sub is_variable_allowed {
+	my ($variable_name, $variable_value) = @_;
```
Comment (r4v5) [inline pagerduty_nagios.pl:175]: for fellow reviewers who don't remember their perl:
> Within a subroutine the array @_ contains the parameters passed to that subroutine. Inside a subroutine, @_ is the default array for the array operators push, pop, shift, and unshift.

Thank you Chris for not using bare array operators in this btw

**Review by r4v5 — APPROVED** [review]
Two typo fixes I don't particularly care about, :+1: from me either way

File: pagerduty_nagios.pl (hunk 3 above)
Comment (cswingler) [reply pagerduty_nagios.pl:175]: it is 2019, i have the bytes to spare to make this actually readable

**Pull request merged**